systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java

```diff
@@ -4,28 +4,254 @@
 import static io.strimzi.systemtest.Constants.REGRESSION;
 import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.CoreMatchers.nullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
 
 @Tag(REGRESSION)
 public class CruiseControlST extends BaseST {
 
     private static final Logger LOGGER = LogManager.getLogger(CruiseControlST.class);
+    private static final String NAMESPACE = "cruise-control-test";
+
+    private static final String CRUISE_CONTROL_NAME = "Cruise Control";
+    private static final int CRUISE_CONTROL_DEFAULT_PORT = 9090;
+    private static final String CRUISE_CONTROL_ENDPOINT = "/kafkacruisecontrol/state";
+    private static final String CRUISE_CONTROL_POD_PREFIX = CLUSTER_NAME + "-cruise-control-";
+
+    private static final String CRUISE_CONTROL_CAPACITY_FILE_PATH = "/tmp/capacity.json";
+    private static final String CRUISE_CONTROL_CONFIGURATION_FILE_PATH = "/tmp/cruisecontrol.properties";
+
+    private static final String CRUISE_CONTROL_METRICS_TOPIC = "strimzi.cruisecontrol.metrics"; // partitions 1 , rf - 1
+    private static final String CRUISE_CONTROL_MODEL_TRAINING_SAMPLES_TOPIC = "strimzi.cruisecontrol.modeltrainingsamples"; // partitions 32 , rf - 2
+    private static final String CRUISE_CONTROL_PARTITION_METRICS_SAMPLES_TOPIC = "strimzi.cruisecontrol.partitionmetricsamples"; // partitions 32 , rf - 2
+
+    @Test
+    void testCruiseControlDeployment()  {
+        String ccStatusCommand = "curl -X GET localhost:" + CRUISE_CONTROL_DEFAULT_PORT + CRUISE_CONTROL_ENDPOINT;
+
+        String ccPodName = PodUtils.getFirstPodNameContaining("cruise-control");
+
+        String result = cmdKubeClient().execInPodContainer(ccPodName, "cruise-control",
+            "/bin/bash", "-c", ccStatusCommand).out();
+
+        LOGGER.info("Verifying that {} is running inside the Pod {} and REST API is available", CRUISE_CONTROL_NAME, ccPodName);
+
+        assertThat(result, not(containsString("404")));
+        assertThat(result, containsString("RUNNING"));
+
+        verifyThatCruiseControlTopicsArePresent();
+    }
+
+    @Test
+    void testUninstallingAndInstallationCruiseControl() {
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, kafka -> {
+            LOGGER.info("Removing Cruise Control to the classic Kafka.");
+            kafka.getSpec().setCruiseControl(null);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), 3, kafkaPods);
```

Comment:
Would be feasible to test that the metrics reporter is not configured anymore in the new Kafka brokers because we removed CC? We could check the Kafka pod log or maybe better the configmap containing the updated broker configuration? Wdyt @kyguy @tomncooper

Reply:
That is a good idea, either or should work fine!